src/drive/mobile/containers/UploadProgression.jsx

```diff
@@ -1,13 +1,87 @@
+import React from 'react'
 import { connect } from 'react-redux'
-import UploadProgression from '../components/UploadProgression'
-
-const mapStateToProps = state => ({
-  message: state.mobile.mediaBackup.currentUpload
-    ? state.mobile.mediaBackup.currentUpload.message
-    : undefined,
-  messageData: state.mobile.mediaBackup.currentUpload
-    ? state.mobile.mediaBackup.currentUpload.messageData
-    : undefined
-})
-
-export default connect(mapStateToProps)(UploadProgression)
+import classnames from 'classnames'
+import styles from '../styles/uploadprogression'
+import { translate } from 'cozy-ui/react/I18n'
+
+const percent = (actual, total) => Math.floor(actual * 100 / total)
+
+const Progress = ({ percent, style, color, background }) => {
+  const containerStyle = {
+    opacity: percent < 100 ? 1 : 0,
+    transition: `${0.4}s opacity}`,
+    transitionDelay: `${percent < 100 ? 0 : 0.4}s`,
+    height: `${2}px`,
+    backgroundColor: background ? 'lightgray' : ''
+  }
+  const barStyle = {
+    display: 'inline-block',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    width: `${percent}%`,
+    maxWidth: '100%',
+    height: `${2}px`,
+    borderRadius: '0 1px 1px 0',
+    transition: `${0.4}s width, ${0.4}s background-color`,
+    backgroundColor: color || 'darkblue',
+    ...style
+  }
+  return (
+    <div style={containerStyle}>
+      <div style={barStyle} />
+    </div>
+  )
+}
+
+const UploadProgression = ({ t, current, total, media }) => {
+  return (
+    <div className={styles['coz-upload-status']}>
+      <Progress percent={percent(current, total)} />
+      {media.filePath ? (
+        <img className={styles['coz-progress-pic']} src={media.filePath} />
```

Comment:
you didn't want to pass the `filePath` as a prop instead of the `media` object? I also just noticed the `img` tag has no `alt`. We don't have any very useful alt title available, but I suppose the file's name is better than noting.

Reply:
I don't know, if `alt` is mandatory. What will be its purpose? A11y? Display `alt` if the `filepath` does not contain a image file?

Reply:
> If we want thumbnails, it is not extraordinary to get everything about the file and not only the filepath. That is a small decision with no side effect.

Ok, that's good enough for me! And yes I was thinking of the `alt` for for accessibility reasons.